### ISSUE TITLE
Uncaught (in promise) Error: Git head failed

### ISSUE THREAD

**Issue by Arcanemagus** [atom/atom#15920]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Opening a repository that is completely new, as in no commits have been made yet, shows an error in the console.

### Steps to Reproduce

1. Create an empty folder.
2. Run `git init` in the folder
3. Open the folder in Atom

**Expected behavior:**

No errors printed to the console.

**Actual behavior:**

An `Uncaught (in promise) Error: Git head failed` error is printed to the Console:

![image](https://user-images.githubusercontent.com/427137/31683065-b12a6646-b330-11e7-9415-95ecf80f62fb.png)

**Reproduces how often:**

100% of the time.

### Versions

```
Atom    : 1.22.0-beta1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0
apm  1.18.8
npm  3.10.10
node 6.9.5 x64
python 2.7.14
git 2.14.2.windows.3
visual studio
```

OS: Windows 10 x64 Enterprise

### Additional Information

If you initialize the repository in Atom itself using the `github` package the same issue happens if you then reload the window.

Originally filed as https://github.com/atom/github/issues/1222.

**Comment by 50Wliu** [comment]
Can confirm - was about to file this myself :).

**Comment by iolsen** [comment]
@BinaryMuse this seems prime for you to investigate this week.

**Comment by maxbrunsfeld** [comment]
This was probably introduced by https://github.com/atom/atom/pull/15750.

**Comment by 50Wliu** [comment]
Fixed by atom/git-utils#84.
